include/clap/ext/draft/preset-load.h

```diff
@@ -2,18 +2,43 @@
 
 #include "../../plugin.h"
 
-static const char CLAP_EXT_PRESET_LOAD[] = "clap.preset-load.draft/0";
+static const char CLAP_EXT_PRESET_LOAD[] = "clap.preset-load.draft/1";
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
 typedef struct clap_plugin_preset_load {
-   // Loads a preset in the plugin native preset file format from a path.
+   // Loads a preset in the plugin native preset file format from a URI. eg:
+   // - "file:///home/abique/.u-he/Diva/Presets/Diva/HS Bass Nine.h2p", load_key: null
+   // - "plugin://<plugin-id>", load_key: <XXX>
+   //
+   // The preset discovery provider defines the uri and load_key to be passed to this function.
+   //
    // [main-thread]
-   bool(CLAP_ABI *from_file)(const clap_plugin_t *plugin, const char *path);
+   bool(CLAP_ABI *from_uri)(const clap_plugin_t *plugin, const char *uri, const char *load_key);
```

Comment:
Is this call synchronous? e.g. if the plugin needs to load large sample files from the user's drive for the preset, do these loads need to finish before the `from_uri` call returns?

Reply:
Presumably the semantics are exactly the same as `clap_plugin_state::load()`.

Reply:
The state extension doesn't document it either way. For now, I've made a mini-extension just for my internal tools (I needed it for unit testing)
```
struct clap_plugin_nakst_ensure_samples_loaded { void (*ensure_samples_loaded)(const clap_plugin_t *plugin); };
#define CLAP_EXT_NAKST_ENSURE_SAMPLES_LOADED "nakst.ensure_samples_loaded"
```

Reply:
Yes this call is supposed to be synchronous, you can still decide to return quickly and finish loading in background if your code supports it.
I think that in  your situation: "loading samples", you can block in `activate()` if the background sample loading isn't finished.

Reply:
If the plugin is already activated, then it is a best effort loading, as if you click on the preset from your GUI.